nginx.py

```diff
@@ -436,49 +436,56 @@ def loads(data, conf=True):
             index += m.end()
             continue
 
-        m = re.compile(r'^\s*location\s*(.*?)\s*{', re.S).search(data[index:])
+        m = re.compile(r'^\s*location\s*([^;]*?)\s*{', re.S).search(data[index:])
         if m:
             l = Location(m.group(1))
             lopen.insert(0, l)
             index += m.end()
             continue
 
-        m = re.compile(r'^\s*if\s*(.*?)\s*{', re.S).search(data[index:])
+        m = re.compile(r'^\s*if\s*([^;].*?)\s*{', re.S).search(data[index:])
```

Comment:
The `.` is still in this `if` block and the following `upstream`, `geo`, `map`, `limit_exception` btw. Forgotten to write it in the previous review.

Reply:
Other than that this looks good to me. Good job! :smile:

Reply:
And for me too @fulder Thanks again for you review.

Reply:
You should probably remove the `.` from the other places as well.

Reply:
I did it hard but I think I did it. ))

Reply:
looks good now +1 from me to merge